x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/Classification.java

```diff
@@ -152,17 +158,37 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     }
 
     @Override
-    public Map<String, Object> getParams() {
+    public Map<String, Object> getParams(Map<String, Set<String>> extractedFields) {
         Map<String, Object> params = new HashMap<>();
         params.put(DEPENDENT_VARIABLE.getPreferredName(), dependentVariable);
         params.putAll(boostedTreeParams.getParams());
         params.put(NUM_TOP_CLASSES.getPreferredName(), numTopClasses);
         if (predictionFieldName != null) {
             params.put(PREDICTION_FIELD_NAME.getPreferredName(), predictionFieldName);
         }
+        String predictionFieldType = getPredictionFieldType(extractedFields.get(dependentVariable));
+        if (predictionFieldType != null) {
+            params.put(PREDICTION_FIELD_TYPE, predictionFieldType);
+        }
         return params;
     }
 
+    private static String getPredictionFieldType(Set<String> dependentVariableTypes) {
+        if (dependentVariableTypes == null) {
+            return null;
+        }
+        if (Types.bool().containsAll(dependentVariableTypes)) {
+            return "bool";
+        }
+        if (Types.discreteNumerical().containsAll(dependentVariableTypes)) {
+            return "int";
```

Comment:
Is it valid for a user to run a categorical job where the dependent variable is a `long` and the two values are too large to be stored within an int mapped value?

Reply:
`int` here maps to an actual type of `int64_t` in the C++ code, which is the equivalent of Java's `long`.  The problem you're referring to was discussed in https://github.com/elastic/ml-cpp/pull/877#discussion_r354204017 and isn't a problem any more technically.  But maybe it is still a problem from a naming perspective.  Renaming this now will mean another C++ change, so the pragmatic solution is to add a comment.